### ISSUE TITLE
Fatal: Layer DepthwiseConv2d is not supported

### ISSUE THREAD

**Issue by divyeshpatel229** [kendryte/nncase#14]
2019-05-03 12:05:56.214925: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
Fatal: Layer DepthwiseConv2d is not supported
NnCase.Converter.Converters.LayerNotSupportedException: Layer DepthwiseConv2d is not supported
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.<Convert>g__ConvertLayer|0_6(Layer layer, <>c__DisplayClass0_0& ) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 51
   at NnCase.Converter.K210.Converters.Stages.Convert.Converter.Convert(Graph graph, QuantizationContext quantizationContext, Int32 weightsBits) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\Stages\Convert\Converter.cs:line 60
   at NnCase.Converter.K210.Converters.GraphToK210Converter.ConvertAsync(Dataset dataset, GraphPlanContext planContext, String outputDir, String prefix, Boolean channelwiseOutput) in C:\projects\nncase\src\NnCase.Converter.K210\Converters\GraphToK210Converter.cs:line 38
   at NnCase.Cli.Program.Main(String[] args) in C:\projects\nncase\src\NnCase.Cli\Program.cs:line 246
   at NnCase.Cli.Program.<Main>(String[] args)

**Comment by Zhengzhongyou** [comment]
Has it been solved?

**Comment by sunnycase** [comment]
Check your DepthwiseConv2d with 3x3 kernel and 2x2 stride, there is a hardware limitation that you must use tf.pad([[0,0],[1,1],[1,1],[0,0]]) to pad your input and the use `valid` padding in your DepthwiseConv2d.

**Comment by Zhengzhongyou** [comment]
您好！您这边可以通过有偿的方式提供一个人脸识别的完整模型么?我们这里可以支付您费用！


------------------ 原始邮件 ------------------
发件人: "sunnycase"<notifications@github.com>;
发送时间: 2019年5月6日(星期一) 中午1:44
收件人: "kendryte/nncase"<nncase@noreply.github.com>;
抄送: "景云亿联-研发"<3030360548@qq.com>; "Comment"<comment@noreply.github.com>; 
主题: Re: [kendryte/nncase] Fatal: Layer DepthwiseConv2d is not supported(#14)




Check your DepthwiseConv2d with 3x3 kernel and 2x2 stride, there is a hardware limitation that you must use tf.pad([[0,0],[1,1],[1,1],[0,0]]) to pad your input and the use valid padding in your DepthwiseConv2d.

**Comment by sunnycase** [comment]
> 
> 您好！您这边可以通过有偿的方式提供一个人脸识别的完整模型么?我们这里可以支付您费用！
> […](#)
> ------------------ 原始邮件 ------------------ 发件人: "sunnycase"<notifications@github.com>; 发送时间: 2019年5月6日(星期一) 中午1:44 收件人: "kendryte/nncase"<nncase@noreply.github.com>; 抄送: "景云亿联-研发"<3030360548@qq.com>; "Comment"<comment@noreply.github.com>; 主题: Re: [kendryte/nncase] Fatal: Layer DepthwiseConv2d is not supported(#14) Check your DepthwiseConv2d with 3x3 kernel and 2x2 stride, there is a hardware limitation that you must use tf.pad([[0,0],[1,1],[1,1],[0,0]]) to pad your input and the use valid padding in your DepthwiseConv2d. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.

你可以联系商务。

**Comment by Zhengzhongyou** [comment]
好的，谢谢！




------------------ 原始邮件 ------------------
发件人: "sunnycase"<notifications@github.com>;
发送时间: 2019年5月7日(星期二) 下午4:48
收件人: "kendryte/nncase"<nncase@noreply.github.com>;
抄送: "景云亿联-研发"<3030360548@qq.com>; "Comment"<comment@noreply.github.com>; 
主题: Re: [kendryte/nncase] Fatal: Layer DepthwiseConv2d is not supported(#14)



 
您好！您这边可以通过有偿的方式提供一个人脸识别的完整模型么?我们这里可以支付您费用！
 …
 ------------------ 原始邮件 ------------------ 发件人: "sunnycase"notifications@github.com; 发送时间: 2019年5月6日(星期一) 中午1:44 收件人: "kendryte/nncase"nncase@noreply.github.com; 抄送: "景云亿联-研发"3030360548@qq.com; "Comment"comment@noreply.github.com; 主题: Re: [kendryte/nncase] Fatal: Layer DepthwiseConv2d is not supported(#14) Check your DepthwiseConv2d with 3x3 kernel and 2x2 stride, there is a hardware limitation that you must use tf.pad([[0,0],[1,1],[1,1],[0,0]]) to pad your input and the use valid padding in your DepthwiseConv2d. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.
  
你可以联系商务。

**Comment by Zhengzhongyou** [comment]
您这边可以给我们推荐一个对你们这个芯片兼容比较好的框架来学习么？谢谢您！

**Comment by divyeshpatel229** [comment]
> Check your DepthwiseConv2d with 3x3 kernel and 2x2 stride, there is a hardware limitation that you must use tf.pad([[0,0],[1,1],[1,1],[0,0]]) to pad your input and the use `valid` padding in your DepthwiseConv2d.

I have used tensorflow detection api. i have no idea how to add padding.

**Comment by divyeshpatel229** [comment]
Hiii i don't have any idea. Please guide me how to do it in tensorflow. I am using tensorflow api. If you have source code of face recognition than please give me.

**Comment by divyeshpatel229** [comment]
> Check your DepthwiseConv2d with 3x3 kernel and 2x2 stride, there is a hardware limitation that you must use tf.pad([[0,0],[1,1],[1,1],[0,0]]) to pad your input and the use `valid` padding in your DepthwiseConv2d.

Where do i need to change these?

**Comment by sunnycase** [comment]
@divyeshpatel229 Your training scripts. You can take a look at https://github.com/kendryte/tensorflow-workspace/blob/master/mobilenetv1/models/mobilenet_v1.py#L50.

**Comment by divyeshpatel229** [comment]
> @divyeshpatel229 Your training scripts. You can take a look at https://github.com/kendryte/tensorflow-workspace/blob/master/mobilenetv1/models/mobilenet_v1.py#L50.

Well i tried this code but i'm getting training issues.

**Comment by divyeshpatel229** [comment]
Running training
use_fixed_image_standardization=1
Epoch: [1][1/5]	Time 6.263	Loss 2.581	Xent 2.581	Accuracy 0.000	Lr 0.10000
Epoch: [1][2/5]	Time 3.446	Loss 3.036	Xent 3.036	Accuracy 0.000	Lr 0.10000
Epoch: [1][3/5]	Time 3.732	Loss 2.259	Xent 2.259	Accuracy 0.125	Lr 0.10000
Epoch: [1][4/5]	Time 3.448	Loss 2.830	Xent 2.830	Accuracy 0.125	Lr 0.10000
Epoch: [1][5/5]	Time 3.520	Loss 2.568	Xent 2.568	Accuracy 0.125	Lr 0.10000
Saving variables
Variables saved in 0.11 seconds
Saving metagraph
Metagraph saved in 0.90 seconds
use_fixed_image_standardization=1
Epoch: [2][1/5]	Time 3.568	Loss 2.425	Xent 2.425	Accuracy 0.000	Lr 0.10000
Epoch: [2][2/5]	Time 3.303	Loss 2.447	Xent 2.447	Accuracy 0.250	Lr 0.10000
Epoch: [2][3/5]	Time 3.349	Loss 3.057	Xent 3.057	Accuracy 0.125	Lr 0.10000
Epoch: [2][4/5]	Time 3.385	Loss 2.354	Xent 2.354	Accuracy 0.000	Lr 0.10000
Epoch: [2][5/5]	Time 3.344	Loss 2.252	Xent 2.252	Accuracy 0.125	Lr 0.10000
Saving variables
Variables saved in 0.02 seconds
use_fixed_image_standardization=1
Epoch: [3][1/5]	Time 3.409	Loss 2.663	Xent 2.663	Accuracy 0.250	Lr 0.10000
Epoch: [3][2/5]	Time 3.321	Loss 2.033	Xent 2.033	Accuracy 0.125	Lr 0.10000
Epoch: [3][3/5]	Time 3.354	Loss 2.506	Xent 2.506	Accuracy 0.375	Lr 0.10000
Epoch: [3][4/5]	Time 3.485	Loss 2.558	Xent 2.558	Accuracy 0.250	Lr 0.10000
Epoch: [3][5/5]	Time 3.374	Loss 2.065	Xent 2.065	Accuracy 0.250	Lr 0.10000
Saving variables
Variables saved in 0.02 seconds
use_fixed_image_standardization=1
Epoch: [4][1/5]	Time 3.440	Loss 2.423	Xent 2.423	Accuracy 0.250	Lr 0.10000
Epoch: [4][2/5]	Time 3.353	Loss 1.999	Xent 1.999	Accuracy 0.250	Lr 0.10000
Epoch: [4][3/5]	Time 3.504	Loss 2.663	Xent 2.663	Accuracy 0.250	Lr 0.10000
Epoch: [4][4/5]	Time 4.096	Loss 1.964	Xent 1.964	Accuracy 0.125	Lr 0.10000
Epoch: [4][5/5]	Time 4.483	Loss 2.157	Xent 2.157	Accuracy 0.375	Lr 0.10000
Saving variables
Variables saved in 0.02 seconds
use_fixed_image_standardization=1
Epoch: [5][1/5]	Time 3.694	Loss 2.321	Xent 2.321	Accuracy 0.250	Lr 0.10000
Epoch: [5][2/5]	Time 3.382	Loss 3.044	Xent 3.044	Accuracy 0.000	Lr 0.10000
Epoch: [5][3/5]	Time 3.375	Loss 1.711	Xent 1.711	Accuracy 0.500	Lr 0.10000
Epoch: [5][4/5]	Time 3.344	Loss 1.617	Xent 1.617	Accuracy 0.500	Lr 0.10000
Epoch: [5][5/5]	Time 3.537	Loss 2.074	Xent 2.074	Accuracy 0.125	Lr 0.10000
Saving variables
Variables saved in 0.02 seconds
use_fixed_image_standardization=1
Epoch: [6][1/5]	Time 3.528	Loss 1.866	Xent 1.866	Accuracy 0.500	Lr 0.10000
Epoch: [6][2/5]	Time 3.369	Loss 1.637	Xent 1.637	Accuracy 0.500	Lr 0.10000
Epoch: [6][3/5]	Time 3.289	Loss 1.830	Xent 1.830	Accuracy 0.125	Lr 0.10000
Epoch: [6][4/5]	Time 3.338	Loss 2.260	Xent 2.260	Accuracy 0.000	Lr 0.10000
Epoch: [6][5/5]	Time 3.294	Loss 1.769	Xent 1.769	Accuracy 0.125	Lr 0.10000
Saving variables


> > @divyeshpatel229 Your training scripts. You can take a look at https://github.com/kendryte/tensorflow-workspace/blob/master/mobilenetv1/models/mobilenet_v1.py#L50.
> 
> Well i tried this code but i'm getting training issues.
